addon/-private/system/model/model.js

```diff
@@ -120,8 +120,8 @@ var Model = Ember.Object.extend(Ember.Evented, {
     @type {Boolean}
     @readOnly
   */
-  hasDirtyAttributes: Ember.computed('currentState.isDirty', function() {
-    return this.get('currentState.isDirty');
+  hasDirtyAttributes: Ember.computed('currentState', function() {
+    return this.get('_internalModel.currentState.isDirty');
```

Comment:
@pangratz I updated the dependent key in the CP here to `currentState`, but the `hasDirtyAttributes` assertion still failed in `1.13` since `this.get('currentState')` is `undefined`. I updated the returned value to `_internalModel.currentState.isDirty` similar to what `retrieveFromCurrentState` is doing. Will this work?